cfgov/v1/static/js/admin/ask_cfpb_tips.js

```diff
@@ -26,22 +28,94 @@
           buttonset.append(buttonElement);
           
           buttonElement.on('click', function (event) {
+
+
             var sel = window.getSelection();
-            var range = sel.getRangeAt(0).cloneRange();
-            var contents = range.cloneContents();
-            var elem = document.createElement("aside");
-            elem.className = "answer-module";
-            elem.appendChild(contents);
-            range.deleteContents();
-            range.insertNode(elem);
-
-            $(range.commonAncestorContainer).find('*').each(function() {
-              if ($(this).is(':empty') || $(this).text() === '') {
-                $(this).remove();
+            var node = sel.baseNode;
+            
+            var tip = $(node).closest( '.' + baseClass )[0];
+
+            // If selection is currently styled as a tip,
+            // remove the tip wrapper element.
+            if ( tip ) {
+              var docFrag = document.createDocumentFragment();
+              while ( tip.firstChild ) {
+                var child = tip.removeChild( tip.firstChild );
+                docFrag.appendChild(child);
+              }
+              tip.parentNode.replaceChild(docFrag, tip);
+              _this.options.editable.setModified();
+              return;
+            }
+
+            // Otherwise, get the selected range. If it starts or ends
+            // with a text node or inline element (a, strong, i), 
+            // expand the selection either to that node's 
+            // start or end bound, or the start or end bound of its parent
+            // if it's not a direct child of the text input div.
+            var range = rangy.getSelection().getRangeAt(0);
+            
+            function expandSelection( node, start ) {
+              while ( node.nodeType === 3 || 
+                      ['A', 'I', 'Strong'].indexOf( node.nodeName ) > -1 ) {
+                var parent = node.parentNode;
+                if ( parent && !parent.classList.contains( textInputClass ) ) {
+                  range[start ? 'setStartBefore' : 'setEndAfter']( parent );
+                  node = parent;
+                } else {
+                  range[start ? 'setStartBefore' : 'setEndAfter']( node );
+                  break;
+                }
+              }
+            }
+
+            expandSelection( range.startContainer, true );
+            expandSelection( range.endContainer );
+
+
+            // Check if range can be surrounded with a wrapper element.
+            // If not, try to find the least common ancestor of the 
+            // start and end containers and expand the start/end of the
+            // range to encompas it so the range can be surrounded.
+            function getDepth( ancestor, node ) {
+              return node === ancestor 
+                          ? 0 : 
+                          $( node ).parentsUntil( common ).length + 1;
+            }
+
```

Comment:
why not just assign `range.canSurroundContents() ` to a variable and make line 112 and `elseif`?